### ISSUE TITLE
More tools for fast object investigation with TimeSeries

### ISSUE THREAD

**Issue by hombit** [lincc-frameworks/tape#233]
Few thoughts of what we could add, feel free to ad more or create sub-issues from these:

- [ ] `TimeSeries.plot()`
`matplotlib` could be an optional dependency. Challenges: mags vs fluxes (mags should be inverted on plot); default colors for passbands
- [ ] `TimeSeries.fold(period)`
Period-fold light curve. Challenge: what is the output? `TimeSeries` with `.time` to be a phase (time % period)?
- [ ] `TimeSeries.filter_band()`
Convert to a single passband timeseries. Should it return a `TimeSeries`' sub-class / super-class instance?
- [ ] `TimeSeries.dropna()`
Remove invalid data points, which could be interpreted wider than just inf/nan values, for example non-positive flux errors are invalid
- [ ] `TimeSeries.__repr__()/__str__()`

**Comment by dougbrn** [comment]
Would be interested to talk about this more as well. We actually have an open ticket (#130) to deprecate TimeSeries in favor of serving users a minimal data structure that hands them light curves as arrays that they can work with as they see fit. The idea being that at the time we didn't think we really had much to offer users on the single-lightcurve visualization/processing front that they wouldn't already have their own tools (or just native pandas in the TimeSeries case) for. That being said, if you think the above would help in your workflows, we can definitely think about how much emphasis to put on the single lightcurve side!

**Comment by hombit** [comment]
From my point of view TAPE is missing tools for data exploration - this issue is basically about it, not about the particular interface. I agree that we cannot and shouldn't offer too much, but I believe it would be nice to offer some basic functionality.

**Comment by dougbrn** [comment]
Yeah, I agree for sure that we want to make sure users have tools for data exploration. I can definitely see opportunity on the plotting side to at least have a short-hand plotter function for users, and obviously they can just use matplotlib directly for any specific needs. Things like `filter_band` are more questionable to me, is something like `TimeSeries.filter_band("g")` much more convenient than the Pandas interface currently available (`TimeSeries.data.loc["g"]`)?

**Comment by hombit** [comment]
`TimeSeries.data.loc["g"]` is cool, we should add it to docs / tutorials!

**Comment by dougbrn** [comment]
The need for a basic .plot function here blocks #404
